x/config/config.go

```diff
@@ -26,6 +26,94 @@ import (
 	"github.com/Jigsaw-Code/outline-sdk/transport/tlsfrag"
 )
 
+// ConfigParser enables the creation of stream and packet dialers based on a config. The config is
+// extensible by registering wrappers for config subtypes.
+type ConfigParser struct {
+	sdWrapers  map[string]WrapStreamDialerFunc
+	pdWrappers map[string]WrapPacketDialerFunc
+}
+
+// NewDefaultConfigParser creates a [ConfigParser] with a set of default wrappers already registered.
+func NewDefaultConfigParser() *ConfigParser {
```

Comment:
Maybe we can have another `NewEmptyConfigParser()`? This can be used by OutlineCLI (`NewEmptyConfigParser().RegisterShadowsocksDialerWrappers()`), because it only expects the `ss://` config.

Reply:
@jyyi1 I also think having `NewEmptyConfigParser()` is useful in general but in the case of Outline CLI why do we want to limit the config to `ss://` only? I am working on a `sock5-over-tls` server that I believe would be useful to folks who are using Outline CLI. 

https://github.com/amircybersec/socks5-over-tls

This way CLI users can also use fragmentation and packet split capability of the Outline SDK.

Reply:
That sounds good to me, we don't have to limit the config type of Outline CLI, we can enable other VPN protocols (such as `socks5` or maybe `vmess`) if we support them. Now we only have `ss`.

Reply:
@jyyi1 BTW, since `ConfigParser` type is exported, one could just do `p := new(ConfigParser)` to create an empty config parser instead of `NewEmptyConfigParser()` but the explicit method can help too.

Reply:
Yeah, the zero struct is valid and can be used.